src/test/java/spaceEngineers/InteractionTest.kt

```diff
@@ -0,0 +1,56 @@
+package spaceEngineers
+
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import spaceEngineers.SeRequest
+import spaceEngineers.commands.*
+import testhelp.checkMockObservation
+import testhelp.environment
+
+class InteractionTest {
+    @Test
+    fun equipToolbarItemTest() = environment {
+        val observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.interact("you", InteractionArgs(InteractionType.EQUIP, 4))))
+        checkMockObservation(observation)
+    }
+
+
+    @Test
+    fun pageAndEquipTest() = environment {
+        val observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.interact("you", InteractionArgs(InteractionType.EQUIP, 4, 2))))
+        checkMockObservation(observation)
+    }
+
+    @Test
+    fun equipAndPlace() = environment {
+        var observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.interact("you", InteractionArgs(InteractionType.EQUIP, 3))))
+        checkMockObservation(observation)
+        observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.interact("you", InteractionArgs(InteractionType.PLACE))))
+        checkMockObservation(observation)
+    }
+
+    @Test
+    fun checkNewBlock() = environment {
+        var observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.interact("you", InteractionArgs(InteractionType.EQUIP, 3))))
+        assertNotNull(observation)
+        observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.interact("you", InteractionArgs(InteractionType.PLACE))))
+        assertNotNull(observation)
+        observation = getSeResponse(SeRequest.command(
+                SeAgentCommand.observe("you", ObservationArgs(ObservationMode.NEW_BLOCKS))))
+        checkMockObservation(observation)
+        assertNotNull(observation.grids)
+        assertTrue(observation.grids.size > 0, "Expecting non-zero block count.")
+        println("Got ${observation.grids.size} blocks.")
+        val firstBlock = observation.grids.first().blocks.first()
+        println("First block max integrity: ${firstBlock.maxIntegrity}")
+        println("First block build integrity: ${firstBlock.buildIntegrity}")
+        println("First block integrity: ${firstBlock.integrity}")
```

Comment:
this is also an outdated version of the test ...perhaps this would be a candidate for one of the "manual game tests"

Reply:
I agree there shouldn't really be sysout in (automated) tests, just checking for parameter values.